src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -651,6 +736,20 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     void TermControl::_KeyDownHandler(winrt::Windows::Foundation::IInspectable const& /*sender*/,
                                       Input::KeyRoutedEventArgs const& e)
     {
+        // If Esc is pressed and the search box is opened, we close the search box
+        if (_searchBox && e.OriginalKey() == winrt::Windows::System::VirtualKey::Escape)

```

Comment:
should the search box be in control of its own `Escape` key handler?

Reply:
is this because we are using PreviewKeyDown? I finally understand why we needed to check `ContainsFocus`

Reply:
(as I mention below: I think this can be handled in `SearchBoxControl`'s `TextBox::KeyDown`, so long as the event is marked `Handled(true)`)